gossip/client/namedtree.go

```diff
@@ -0,0 +1,113 @@
+package client
+
+import (
+	"context"
+	"crypto/ecdsa"
+	"strings"
+
+	"github.com/quorumcontrol/chaintree/chaintree"
+	"github.com/quorumcontrol/chaintree/nodestore"
+	"github.com/quorumcontrol/messages/v2/build/go/transactions"
+	"github.com/quorumcontrol/tupelo-go-sdk/consensus"
+)
+
+type Generator struct {
+	Namespace string
+	Client    *Client
+}
+
+type NamedChainTree struct {
+	Name      string
+	ChainTree *consensus.SignedChainTree
+	Client    *Client
+
+	genesisKey *ecdsa.PrivateKey
+	nodeStore  nodestore.DagStore
+	owners     []string
```

Comment:
this will also be saved in the chaintree itself so it's a little odd in the struct I think.